src/main/kotlin/BUILD

```diff
@@ -26,14 +27,33 @@ kt_bootstrap_library(
     visibility = ["//src/test/kotlin/io/bazel/kotlin/builder:__subpackages__"],
 )
 
-# The builder artifact.
+java_binary(
+    name = "builder_raw",
+    create_executable = False,
+    runtime_deps = ["//src/main/kotlin/io/bazel/kotlin/builder"],
+)
+
+# The builder artifact. Shaded to ensure that libraries it uses are not leaked to
+# the code it's running against (e.g. dagger)
+jar_jar(
+    name = "builder_jar_jar",
+    input_jar = ":builder_raw_deploy.jar",
+    rules = "shade.jarjar",
+)
+
 java_binary(
     name = "builder",
     data = [
         ":compiler_lib.jar",
         "@com_github_jetbrains_kotlin//:lib/kotlin-compiler.jar",
     ],
     main_class = "io.bazel.kotlin.builder.KotlinBuilderMain",
+    runtime_deps = [":builder_jar_jar"],
+    jvm_flags = [
+        "-XX:+IgnoreUnrecognizedVMOptions",
+        "--add-opens=java.base/java.nio=ALL-UNNAMED",
+        "--add-opens=java.base/java.lang=ALL-UNNAMED",
+        "--add-opens=jdk.jdeps/com.sun.tools.jdeps=ALL-UNNAMED",
```

Comment:
No, this is about JDK9+, because some of the things this depends on are "unsafe" APIs with warnings (protobuf is especially an issue) so this declares that these are opened on purpose, so we don't get warnings spam.

Reply:
SGTM